src/webpack/plugin/SitePlugin.js

```diff
@@ -0,0 +1,287 @@
+    siteRegistry[pathPrefix] = {
+      path: pathPrefix,
+      url: this.url,
+      title: titleConfig.title,
+      entry: this.entryKey,
+    };
+  }
+
+  static applyOneTimeSetup({
+    compiler,
+    sourceFolder,
+    distributionFolder,
+    basename,
+  }) {
+    if (oneTimeSetupComplete) {
+      return;
+    }
+    oneTimeSetupComplete = true;
+
+    compiler.options.entry = {
+      ...compiler.options.entry,
+      rewriteHistory: '@cerner/terra-dev-site/lib/browser-router-redirect/rewriteHistory',
+      redirect: '@cerner/terra-dev-site/lib/browser-router-redirect/redirect',
+    };
+
+    // MODULE
+    const mdxLoader = getMdxLoader(compiler.options.output.publicPath);
+    compiler.options.module.rules = [{
+      // Drop loaders in a 'one of' block to avoid the original loaders applying on top of the new loaders.
+      // Only the first loader will apply and no others.
+      oneOf: [{
+        test: /\.mdx$/,
+        use: [
+          babelLoader,
+          mdxLoader,
+        ],
+      }, {
+        test: /\.md$/,
+        oneOf: [
+          {
+            // Use MDX to import any md files imported from an mdx file.
+            issuer: [
+              /\.mdx?$/,
+              /entry\.template$/,
+            ],
+            use: [
+              babelLoader,
+              mdxLoader,
+            ],
+          },
+          {
+            use: 'raw-loader',
+          },
+        ],
+      }, {
+        resourceQuery: '?dev-site-codeblock',
+        // this bypasses the default json loader
+        type: 'javascript/auto',
+        use: [
+          babelLoader,
+          mdxLoader,
+          {
+            loader: 'devSiteCodeblock',
+            options: {
+              resolveExtensions: compiler.options.resolve.extensions,
+            },
+          },
+        ],
+      }, {
+        resourceQuery: '?dev-site-example',
+        use: [
+          babelLoader,
+          'devSiteExample',
+        ],
+      }, {
+        test: /\.json$/,
+        // this bypasses the default json loader
+        type: 'javascript/auto',
+        resourceQuery: '?dev-site-package',
+        use: [
+          babelLoader,
+          'devSitePackage',
+        ],
+      }, {
+        resourceQuery: '?dev-site-props-table',
+        use: [
+          babelLoader,
+          {
+            loader: 'devSitePropsTable',
+            options: {
+              mdx: mdxOptions(compiler.options.output.publicPath),
+              resolveExtensions: compiler.options.resolve.extensions,
+            },
+          },
+        ],
+      },
+      // Spread the original loaders. These will be applied if all above loaders fail.
+      ...compiler.options.module.rules,
+      ],
+    }];
+
+    // RESOLVE
+    // If plugins is not defined, define it.
+    if (!compiler.options.resolve.plugins) {
+      compiler.options.resolve.plugins = [];
+    }
+
+    // If a mono repo, update the rootDirectories to include all the packages.
```

Comment:
This it so allow for a package to alias itself right?

Reply:
yep